p4-16/spec/P4-16-spec.mdk

```diff
@@ -1639,6 +1642,60 @@ directions:
   including values for the directionless parameters. The directionless
   parameters in this case behave like `in` parameters.
   See Section [#sec-invoke-actions] for further details.
+- Default parameter values are only allowed for 'in' or direction-less parameters;
+  these values must evaluate to compile-time constants.
+
+### Optional parameters
+
+A parameter that is annotated with the `@optional` annotation is
+optional: the user may supply no value for that parameter in an
```

Comment:
"May supply no value for the parameter" --> "May omit the value for that parameter"?